### PR TITLE
nodePackages.postcss-cli: add passthru.tests

### PR DESCRIPTION
**Pull request by Luflosi**
###### Motivation for this change
See #130027.

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [x] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes (or backporting 21.05 Relase notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by symphorien** [comment]
This could be run as installCheckPhase instead.

**Comment by roberth** [comment]
> This could be run as installCheckPhase instead.

`installCheckPhase` doesn't have a clean environment. The lack of `node_modules` or related environment variables in the extra derivation is an important feature.

File: pkgs/development/node-packages/package-tests/postcss-cli.nix [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+    # get version of installed program and compare with package version
+    claimed_version="$(${postcss-cli}/bin/postcss --version)"
+    if [[ "$claimed_version" != "${version}" ]]; then
+      echo "Error: program version does not match package version ($claimed_version != ${version})"
+      exit 1
+    fi
+
+    # run basic help command
+    ${postcss-cli}/bin/postcss --help > /dev/null
+
```
Comment (roberth) [inline pkgs/development/node-packages/package-tests/postcss-cli.nix:18]: I'd like to see a tiny amount of css being processed, just to exercise the essential code paths and see that the libraries integrate correctly.

File: pkgs/development/node-packages/package-tests/postcss-cli.nix [hunk 2]
```diff
@@ -0,0 +1,21 @@
+{ runCommand, postcss-cli }:
+
+let
+  inherit (postcss-cli) packageName version;
+in
+
+runCommand "${packageName}-tests" { meta.timeout = 3; }
```
[the remark this answers is not in the document]
Comment (roberth) [reply pkgs/development/node-packages/package-tests/postcss-cli.nix:7]: I see this was suggested by the docs. I'll update those.

[the remark this answers is not in the document]
Comment (Luflosi) [reply pkgs/development/node-packages/package-tests/postcss-cli.nix]: Thank you. I implemented your suggestion.

File: pkgs/development/node-packages/package-tests/postcss-cli.nix (hunk 1 above)
Comment (Luflosi) [reply pkgs/development/node-packages/package-tests/postcss-cli.nix:18]: AFAICT, postcss-cli has no built-in functionality to process CSS, everything seems to be done via plugins. I was planning to add autoprefixer in a different PR. If you like, I can make that PR and when it is accepted, change this PR to test postcss-cli with autoprefixer.

Comment (roberth) [reply pkgs/development/node-packages/package-tests/postcss-cli.nix:18]: Sounds good to me.

Comment (Luflosi) [reply pkgs/development/node-packages/package-tests/postcss-cli.nix:18]: I made the other PR.

Comment (Luflosi) [reply pkgs/development/node-packages/package-tests/postcss-cli.nix:18]: The other PR has been accepted and I updated this PR with a simple autoprefixer test.

File: pkgs/development/node-packages/default.nix [hunk 3]
```diff
@@ -276,6 +276,11 @@ let
           --prefix NODE_PATH : ${self.postcss}/lib/node_modules \
           --prefix NODE_PATH : ${self.autoprefixer}/lib/node_modules
       '';
+      passthru.tests = {
+        simple-execution = pkgs.callPackage ./package-tests/postcss-cli.nix {
+          inherit (self) postcss-cli;
+        };
+      };
```
Comment (roberth) [inline pkgs/development/node-packages/default.nix:283]: It is also relevant for the autoprefixer package. You could add it there as well.

**Review by roberth — APPROVED** [review]
Great work!

**Pull request merged**